lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -68,6 +68,12 @@
             - Toggle that controls if the machine is allocated/deallocated, only useful with state='present'.
         default: True
         type: bool
+    generalized:
+        description:
+            - Use with state 'present' to generalize the machine. Set to true to generalize the machine.
+            - Please note that this operation is irreversible.
```

Comment:
i think we need stronger message to user, that once generalized, the vm is not usable.